src/components/NumberInput.tsx

```diff
@@ -0,0 +1,147 @@
+import { useState } from "react";
+import { Button, Stack, TextField } from "@mui/material";
+
+const Modes = {
+  /** Natural numbers from 0 to inf */
+  NATURAL: /^([0-9]+)$/,
```

Comment:
We've said (not sure to what extend we're doing this) that we'll follow AirBnB JS style guide. I think that would call for camelCase for these keys so
```
const Modes = {
  natural: ....,
  integer: ..., // etc
}
```